Lib/functools.py

```diff
@@ -1122,3 +1117,26 @@ def __get__(self, instance, owner=None):
         return val
 
     __class_getitem__ = classmethod(GenericAlias)
+
+def _warn_kwargs(func):
```

Comment:
To reduce the diff, can't you put it after `_initial_missing`? or is there some reason we can't put it there?

Reply:
Unfortunately, we can't.
If I place `_warn_kwargs` right after `_initial_missing` and decorate `reduce` with `@_warn_kwargs`, it will raise an error:
```python
Traceback (most recent call last):
  File "/home/eclips4/programming/programming-languages/cpython/./Tools/build/check_extension_modules.py", line 20, in <module>
    import argparse
  File "/home/eclips4/programming/programming-languages/cpython/Lib/argparse.py", line 89, in <module>
    import re as _re
  File "/home/eclips4/programming/programming-languages/cpython/Lib/re/__init__.py", line 127, in <module>
    import functools
  File "/home/eclips4/programming/programming-languages/cpython/Lib/functools.py", line 258, in <module>
    @_warn_kwargs
  File "/home/eclips4/programming/programming-languages/cpython/Lib/functools.py", line 242, in _warn_kwargs
    @wraps(func)
  File "/home/eclips4/programming/programming-languages/cpython/Lib/functools.py", line 79, in wraps
    return partial(update_wrapper, wrapped=wrapped,
NameError: name 'partial' is not defined
```

At the moment of calling `wraps` `partial`(Line 390) isn't present in global scope.

Reply:
Arf, I see. Then maybe a small comment saying why we are importing this at the bottom (though I don't know what to write =/)

Reply:
I just wrote that the import should be moved back to Python 3.16 (immediately after the reduce definition), without explaining the exact reason, because it is hard to phrase it.